pkg/controller/node/node_controller.go

```diff
@@ -265,10 +265,18 @@ func (ctrl *Controller) getPoolForNode(node *corev1.Node) (*mcfgv1.MachineConfig
 
 	var pools []*mcfgv1.MachineConfigPool
 	for _, p := range pl {
+		// first get MachineSelector
 		selector, err := metav1.LabelSelectorAsSelector(p.Spec.MachineSelector)
```

Comment:
ok lol, I just realized that every call to `metav1.LabelSelectorAsSelector(` will return `label.Nothing()` which is obviously not nil in the check below (and elsewhere) and this is making our e2e to fail.

The right check should be:

```go
if p.Spec.MachineSelector == nil {
...
}
```

Reply:
if you're wondering the installation goes through because the initial node annotations are loaded from disk (laid down by the bootstrap process so there's no node controller involved which is failing here after cluster is up and running)

Reply:
ahhh ok

Reply:
done